lib/index.js

```diff
@@ -2850,12 +2858,16 @@ addToplevelMethod('pipeline', function (/*through...*/) {
         });
     });
 
+    wrapper.writable = true;
+
     wrapper.write = function (x) {
         return start.write(x);
     };
 
     wrapper.end = function () {
-        return start.end();
+        var result = start.end();
+        wrapper._onEnd();
```

Comment:
(`this.writable`, I assume)
In addition to the `.writable = false` already in `_onEnd`? I thought the latter would be sufficient.